docs/src/unbound_args.md

```diff
@@ -0,0 +1,34 @@
+# Unbound Type Parameters
+
+An unbound type parameter is a type parameter with a `where`,
+that does not occur in the signature of some dispatch of the method.
+
+## Examples
+
+The following methods each have `T` as an unbound type parameter:
+
+```julia
+f(x::Int) where {T} = do_something(x)
+
+g(x::S) where {S < : Number, T <: Number} = do_something(x)
+
+h(x::T...) where {T} = do_something(x)
+```
+
+In the cases of `f` and `g` above, the unbound type parameter `T` is neither
+present in the signature of the methods nor as a bound of another type parameter.
+Here, the type parameter `T` can be removed without changing any semantics.
+
+For signatures with `Vararg` (cf. `h` above), the type parameter is unbound for the 
+zero-argument case (e.g. `h()`). A possible fix would be to replace `h` by two
+methods
+```julia
+h() = do_something(T[])
+h(x1::T, x2::T...) = do_something(T[x1, x2...])
+```
```

Comment:
I think the example can be rewritten like this.

* Example functions `f` and `g` are really similar, so `g` can be removed.
* Use `@repl` block.
* Use `println(T)` instead of `do_something(x)` in `g` for tangible code.
* Add an error example with `ERROR: UndefVarError: `T` not defined`.

````suggestion
```@repl
f(x::Int) where {T} = do_something(x)
g(x::T...) where {T} = println(T)
```

In the cases of `f` above, the unbound type parameter `T` is neither
present in the signature of the methods nor as a bound of another type parameter.
Here, the type parameter `T` can be removed without changing any semantics.

For signatures with `Vararg` (cf. `g` above), the type parameter is unbound for the 
zero-argument case (e.g. `g()`).

```@repl
g(1.0, 2.0)
g(1)
g()
```

A possible fix would be to replace `g` by two methods.

```@repl
h() = println(Int)  # Defaults to `Int`
h(x1::T, x2::T...) where {T}= println(T)
h(1.0, 2.0)
h(1)
h()
```
````

Reply:
````suggestion
```@repl
f(x::Int) where {T} = do_something(x)
g(x::T...) where {T} = println(T)
```

In the cases of `f` above, the unbound type parameter `T` is neither
present in the signature of the methods nor as a bound of another type parameter.
Here, the type parameter `T` can be removed without changing any semantics.

For signatures with `Vararg` (cf. `g` above), the type parameter is unbound for the 
zero-argument case (e.g. `g()`).

```@repl
g(1.0, 2.0)
g(1)
g()
```

A possible fix would be to replace `g` by two methods.

```@repl
g() = println(Int)  # Defaults to `Int`
g(x1::T, x2::T...) where {T} = println(T)
g(1.0, 2.0)
g(1)
g()
```
````